modules/media_riddle_marketplace/src/RiddleMediaService.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\media_riddle_marketplace;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\riddle_marketplace\RiddleFeedServiceInterface;
+
+/**
+ * Class RiddleFeedService.
+ *
+ * @package Drupal\riddle_marketplace
+ */
+class RiddleMediaService implements RiddleMediaServiceInterface {
+
+  /**
+   * The riddle feed service.
+   *
+   * @var \Drupal\riddle_marketplace\RiddleFeedServiceInterface
+   */
+  protected $feedService;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The database service.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * Riddle Media Service.
+   *
+   * Constructor.
+   *
+   * @param \Drupal\riddle_marketplace\RiddleFeedServiceInterface $feedService
+   *   Riddle Feed service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Database\Connection $database
+   *   The database service.
+   */
+  public function __construct(RiddleFeedServiceInterface $feedService, EntityTypeManagerInterface $entityTypeManager, Connection $database) {
+    $this->feedService = $feedService;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->database = $database;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createMediaEntities() {
```

Comment:
If I'm not wrong this is same functionality as: ```\Drupal\media_riddle_marketplace\Controller\RiddleImportController::content```. Would be possible to have one code base for that?